### PR TITLE
Load files using mmap() when possible

### PR DESCRIPTION
**Pull request by hughsie**
This reduces our memory use for large firmware by about 30%.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@superm1 I'm not 100% on the `FuChunker` name -- `FuChunkArray`, `FuChunkGenerator` or `FuChunkBuilder` are alternatives.

File: libfwupdplugin/fu-bytes.c [hunk 1]
```diff
@@ -68,16 +68,33 @@ fu_bytes_set_contents(const gchar *filename, GBytes *bytes, GError **error)
...
-	g_debug("reading %s with %" G_GSIZE_FORMAT " bytes", filename, len);
-	return g_bytes_new_take(data, len);
+	/* try as a mapped file, falling back to reading it as a blob instead */
+	mapped_file = g_mapped_file_new(filename, FALSE, &error_local);
+	if (mapped_file == NULL) {
+		gchar *data = NULL;
+		gsize len = 0;
+		if (!g_file_get_contents(filename, &data, &len, error))
+			return NULL;
+		g_debug("failed to read as mapped file, so reading %s of size 0x%x: %s",
```
Comment (superm1) [inline libfwupdplugin/fu-bytes.c:84]: shouldn't you be using `G_GSIZE_FORMAT`?

**Comment by superm1** [comment]
> @superm1 I'm not 100% on the `FuChunker` name -- `FuChunkArray`, `FuChunkGenerator` or `FuChunkBuilder` are alternatives.

I vote `FuChunkArray`

File: libfwupdplugin/fu-bytes.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-bytes.c:84]: do you know how to coerce G_GSIZE_FORMAT to display in hex format?

**Comment by hughsie** [comment]
I've move the chunk array stuff to https://github.com/fwupd/fwupd/pull/6183

File: libfwupdplugin/fu-bytes.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-bytes.c:84]: Doh, it's easier than I thought. Fixed.

**Pull request merged**